### PR TITLE
allow to skip server installation

### PR DESCRIPTION
**Pull request by nsidnev**
Hi! At the moment, this action is mostly useful for the EdgeDB team only because it involves a mandatory server installation. Usually, when I use GitHub Actions, I just want to run the databases and other services in separate containers managed by GitHub so that it takes care of starting them up, shutting them  down on exit, etc and so I don't have to write additional code to start the server in tests.

As a result, this PR changes how the action works, making the server installation optional by default. I understand that due to following semver specification it will be necessary to release a new major version of the action. Therefore, as an option, I can change the code to add a third parameter `skip-server-installation` as a boolean value. Although it still won't be very convenient IMO. The current approach is based on the [erlef/setup-beam](https://github.com/erlef/setup-beam) action which installs some required stuff along with some additional ones (like elixir or gleam languages).

I tested this code on my project and it worked as I would expect it to work.
https://github.com/nsidnev/edgedb-elixir/blob/25bcc0d1b7af8090a214086470bd9544f0f97566/.github/workflows/tests.yml#L35
https://github.com/nsidnev/edgedb-elixir/runs/3445532295

If there is something in this change that is not what you would like, then I'll be happy to fix it to make it better :)

Refs #10
Refs https://github.com/edgedb/edgedb/discussions/2549#discussioncomment-777809

### CODE REVIEW

**Comment by tailhook** [comment]
Do you want to install just the CLI? Maybe it's better to have `edgedb/setup-edgedb-cli` instead?

**Comment by tailhook** [comment]
I also think that `edgedb/setup-edgedb-cli` could also do `instance link` and `project link` automatically, because although your current solution is very neat, we want to make it simpler.

**Comment by elprans** [comment]
Using `edgedb/setup-edgedb-cli` from action steps wouldn't be very convenient, I think.  

It's fine if we allow disabling either the CLI or the server installation in the action.  I'd use `'none'` as the opt-out indicator, though, to keep compatibility with current behavior.

All that said, I have a plan to update `setup-edgedb` to also support creating/starting instances and doing project linking.  Using github's "service" stuff is neat, but it doesn't work on all platforms and `setup-edgedb` would.

**Comment by nsidnev** [comment]
> Do you want to install just the CLI? Maybe it's better to have edgedb/setup-edgedb-cli instead?

Hmm... I'm not sure about that. Well, spliting action into 2 separate actions could definitely be a way to solve my case, but IMO having a single action would be more convenient. And even if they need to be separated, then as user I'll expect `edgedb/setup-edgedb` action in my pipeline to configure all necessary environment such as CLI, EdgeDB project, linked server instance and so on. And then if I need something more low-level then I can use another action for the server. But again it's just IMO :)

> I also think that edgedb/setup-edgedb-cli could also do instance link and project link automatically, because although your current solution is very neat, we want to make it simpler.

This is actually a very good idea! The lack of this stuff was the original reason why I opened an issue about the non-interactive `edgedb project init` command. The current approach I'm using in my driver and another project is just a temporary hack and it would be great to be able to improve it!

So, summarizing above. I'd rather expect `edgedb/setup-edgedb` to set up GitHub Actions environment as an EdgeDB user rather than as an EdgeDB developer. I see this action more as a sort of EdgeDB CLI adapter or something similar in the CI/CD provider.

**Comment by nsidnev** [comment]
> It's fine if we allow disabling either the CLI or the server installation in the action. I'd use 'none' as the opt-out indicator, though, to keep compatibility with current behavior.

Sorry, I'm not sure I understand you here. Should I just change the current default for `server-version` to `none` instead of empty string (`""`) and skip server installation in that case? Or would you like to skip this change and work on an improved version of this action?

> All that said, I have a plan to update setup-edgedb to also support creating/starting instances and doing project linking. Using github's "service" stuff is neat, but it doesn't work on all platforms and setup-edgedb would.

This sounds very interesting. If this is not very difficult for you, could you please explain your plan in more detail? Maybe in a separate GitHub discussion?

**Comment by elprans** [comment]
> Should I just change the current default for `server-version` to `none` instead of empty string (`""`) and skip server installation in that case?

Yes, this is what I meant.

> Or would you like to skip this change and work on an improved version of this action?

I think an option to skip installation of either component has merit on its own.

> This sounds very interesting. If this is not very difficult for you, could you please explain your plan in more detail? Maybe in a separate GitHub discussion?

It's on my TODO list (albeit not at the very top).  From a CI user's perspective there will simply be more options to pass to `setup-edgedb` to:

1. Automatically bootstrap and start an instance with the specified name.
2. If an `edgedb.toml` is present, link the project with the instance and optionally run project migrations and/or CI fixtures.

Perhaps the above behavior should be the default, because that's what you usually want.

The end result is that your tests and CLI invocations would magically work without any further setup at all.

**Comment by nsidnev** [comment]
> Yes, this is what I meant.

Done :)

> Perhaps the above behavior should be the default, because that's what you usually want.

Actually, changing action to do these 2 steps doesn't seem like a big deal, so I can implement them in next separate (or several separate) PR.
The only blocker at the moment is the lack of `edgedb project link` command, but it also doesn't seem to be difficult to implement (I hope so). So if @tailhook is busy with other tasks, I can take that as well.
This behavior will improve workflows in my current and some future EdgeDB projects, so I'll be happy to help with that.

File: action.yml [hunk 1]
```diff
@@ -15,8 +15,9 @@ inputs:
     required: false
     description: >
       Version of EdgeDB server to install, or "stable" for the latest
-      stable release, or "nightly" for the latest nightly build.
-      Defaults to "stable".
+      stable release, or "nightly" for the latest nightly build,
+      or "none" to skip installaion.
+    default: 'none'
```
Comment (elprans) [inline action.yml:20]: Let's continue to default to 'stable' to preserve current behavior.

**Review by elprans — APPROVED** [review]

**Comment by nsidnev** [comment]
This change doesn't look like something that can cause so much troubles with CI tests :) 
I just hope that after merging upstream, which also fixes the install command, all errors go away.

**Comment by elprans** [comment]
> This change doesn't look like something that can cause so much troubles with CI tests :)

The issue seems to be that the server `PATH` is now added _after_ the CLI `PATH` and it seems like the server `/bin` folder erroneously includes the CLI stub (it should only do so for dev builds).  So, swapping the order of `core.addPath` here back to the original should make CI pass.

**Comment by elprans** [comment]
Thanks for the PR!

> Actually, changing action to do these 2 steps doesn't seem like a big deal, so I can implement them in next separate (or several separate) PR.

That contribution would be most welcome!

**Pull request merged**